sensors/dependencies/filter.go

```diff
@@ -56,50 +56,78 @@ func filterEvent(filter *v1alpha1.EventDependencyFilter, event *v1alpha1.Event)
 	return timeFilter && ctxFilter && dataFilter, err
 }
 
-// applyTimeFilter checks the eventTime against the timeFilter:
-// 1. the eventTime is greater than or equal to the start time
-// 2. the eventTime is less than the end time
-// returns true if 1 and 2 are true and false otherwise
+// filterTime checks the eventTime falls into time range specified by the timeFilter.
+// Start is inclusive, and Stop is exclusive.
+//
+// if Start < Stop: eventTime must be in [Start, Stop)
+//
+//   0:00        Start       Stop        0:00
+//   ├───────────●───────────○───────────┤
+//               └─── OK ────┘
+//
+// if Stop < Start: eventTime must be in [Start, Stop@Next day)
+//
+// this is equivalent to: eventTime must be in [0:00, Stop) or [Start, 0:00@Next day)
+//
+//   0:00                    Start       0:00       Stop                     0:00
+//   ├───────────○───────────●───────────┼───────────○───────────●───────────┤
+//                           └───────── OK ──────────┘
+//
+//   0:00        Stop        Start       0:00
+//   ●───────────○───────────●───────────○
+//   └─── OK ────┘           └─── OK ────┘
+//
+// if only Start is specified: eventTime must be in [Start, 0:00@Next day)
+//
+//   0:00        Start                   0:00
+//   ├───────────●───────────────────────┤
+//               └───────── OK ──────────┘
+//
+// if only Stop is specified: eventTime must be in [0:00, Stop)
+//
+//   0:00                    Stop        0:00
+//   ├───────────────────────○───────────┤
+//   └───────── OK ──────────┘
 func filterTime(timeFilter *v1alpha1.TimeFilter, eventTime time.Time) (bool, error) {
-	if timeFilter != nil {
-		utc := time.Now().UTC()
-		currentTime := time.Date(utc.Year(), utc.Month(), utc.Day(), 0, 0, 0, 0, time.UTC).Format(common.StandardYYYYMMDDFormat)
-
-		if timeFilter.Start != "" && timeFilter.Stop != "" {
-			startTime, err := time.Parse(common.StandardTimeFormat, fmt.Sprintf("%s %s", currentTime, timeFilter.Start))
-			if err != nil {
-				return false, err
-			}
-			startTime = startTime.UTC()
+	if timeFilter == nil {
+		return true, nil
+	}
 
-			stopTime, err := time.Parse(common.StandardTimeFormat, fmt.Sprintf("%s %s", currentTime, timeFilter.Stop))
-			if err != nil {
-				return false, err
-			}
-			stopTime = stopTime.UTC()
+	if timeFilter.Start != "" && timeFilter.Stop != "" {
```

Comment:
Parse both `Start` and `Stop` before doing the real `if, else` logic to simplify the code.

Reply:
Is this what you mean? 1d37e962abc831ed3af3c873409fd84d5385df38

Reply:
How about this:

```
var startTime, stopTime time.Time
var err Error
if timeFilter.Start != "" {
  startTime, err = common.ParseTime(timeFilter.Start, eventTime)
  if err != nil {
    return false, err
  }
}
// same for timeFilter.Stop
xxxx

// real logic
if timeFilter.Start != "" && timeFilter.Stop != "" {
  .....
```

Reply:
Ah I see!

Reply:
✅ dccbd181c189c01c87f51dc6a5d1c076a5c9a3d8